src/SecurityChecks.js

```diff
@@ -0,0 +1,122 @@
+      }
+    };
+    const promises = [
+      getResultForSecurityCheck('CLP', checkCLP),
+      getResultForSecurityCheck('ServerConfig', checkServerConfig),
+      getResultForSecurityCheck('Files', checkFiles),
+    ];
+    if (
+      options.databaseAdapter.getSecurityLogs &&
+      typeof options.databaseAdapter.getSecurityLogs === 'function'
+    ) {
+      promises.push(getResultForSecurityCheck('Database', options.databaseAdapter.getSecurityLogs));
+    }
+    await Promise.all(promises);
+    response.Total = totalWarnings;
+    return { response };
+  } catch (error) {
+    return { error: { code: 1, error: error.message || 'Internal Server Error.' } };
+  }
+}
+
+async function checkCLP(req) {
+  const options = req.config || req;
+  const schema = await options.database.loadSchema();
+  const all = await schema.getAllClasses();
+  const clpWarnings = {};
+  for (const field of all) {
+    const className = field.className;
+    const clp = field.classLevelPermissions;
+    const thisClassWarnings = clpWarnings[className] || [];
+    if (!clp) {
+      thisClassWarnings.push({
+        title: `No Class Level Permissions on ${className}`,
+        message: `Any client can create, find, count, get, update, delete, or add field on ${className}. This allows an attacker to create new objects or fieldNames without restriction and potentially flood the database. Set CLPs using Parse Dashboard.`,
+      });
+      clpWarnings[className] = thisClassWarnings;
+      continue;
+    }
+    const keys = ['find', 'count', 'get', 'create', 'update', 'delete', 'addField'];
+    for (const key of keys) {
+      const option = clp[key];
+      if (className === '_User' && key === 'create') {
+        continue;
+      }
+      if (!option || option['*']) {
+        thisClassWarnings.push({
+          title: `Unrestricted access to ${key}.`,
+          message: `Any client can ${key} on ${className}.`,
+        });
+      } else if (Object.keys(option).length != 0 && key === 'addField') {
+        thisClassWarnings.push({
+          title: `Certain users can add fields.`,
+          message: `Certain users can add fields on ${className}. This allows these users to create new fieldNames and potentially flood the schema. Set CLPs using Parse Dashboard.`,
+        });
+      }
+    }
+    clpWarnings[className] = thisClassWarnings;
+  }
+  return clpWarnings;
+}
+function checkServerConfig(req) {
+  const options = req.config || req;
+  const warnings = [];
+  if (options.allowClientClassCreation) {
+    warnings.push({
+      title: 'Client class creation allowed',
+      message:
+        'Clients are currently allowed to create new classes. This allows an attacker to create new classes without restriction and potentially flood the database. Change the Parse Server configuration to allowClientClassCreation: false.',
+    });
+  }
+  if (!options.masterKey.match('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{14,})')) {
+    warnings.push({
```

Comment:
same as db password but here we can check the length i think, since the Parse API is inevitably exposed to public internet. (Note DB password do not need the length check since in modern cloud computing database is only accessible within an internal network (ex: kubernetes), and developers can choose to use short db passwords)